File: src/main/java/org/oewntk/model/TagCount.java
/*
 * Copyright (c) 2021-2021. Bernard Bou.
 */

package org.oewntk.model;

import java.io.Serializable;

/**
 * Tag count
 */
public class TagCount implements Serializable
{
	private final int senseNum;

	private final int count;

	/**
	 * Constructor
	 *
	 * @param senseNum sense num
	 * @param count    tag count
	 */
	public TagCount(final int senseNum, final int count)
	{
		this.senseNum = senseNum;
		this.count = count;
	}

	/**
	 * Get tag count
	 *
	 * @return tag count
	 */
	public int getCount()
	{
		return count;
	}

	/**
	 * Get sense number for that sensekey (index of sense in pos)
	 *
	 * @return sense num
	 */
	public int getSenseNum()
	{
		return senseNum;
	}

	// stringify

	@Override
	public String toString()
	{
		return Integer.toString(getCount());
	}
}

File: src/main/java/org/oewntk/model/TagCount.kt
/*
 * Copyright (c) 2021-2024. Bernard Bou.
 */
package org.oewntk.model

import java.io.Serializable

/**
 * Tag count
 *
 * @param senseNum    sense num
 * @param count       tag count
 * @property senseNum sense num
 * @property count    tag count
 */
class TagCount(
    private val senseNum: Int,
    val count: Int,

    ) : Serializable {

    override fun toString(): String {
        return count.toString()
    }
}
